From d5cecbdfff88a2b69f02410b8d78001753d45084 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Tue, 11 Mar 2014 18:41:45 +0800
Subject: [PATCH 0320/1074] ENGR00302869-1 ARM: dts: imx6qdl: add cfg_clk for
 MIPI CSI2

MIPI CSI2 depends on this clock to work.
This patch also updates the binding document.

Signed-off-by: Robby Cai <R63905@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 .../devicetree/bindings/video/fsl,mipi-csi2.txt    |    4 ++--
 arch/arm/boot/dts/imx6qdl.dtsi                     |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/video/fsl,mipi-csi2.txt b/Documentation/devicetree/bindings/video/fsl,mipi-csi2.txt
index 4b50280..d10f237 100644
--- a/Documentation/devicetree/bindings/video/fsl,mipi-csi2.txt
+++ b/Documentation/devicetree/bindings/video/fsl,mipi-csi2.txt
@@ -27,8 +27,8 @@ for SOC imx6qdl.dtsi:
 		compatible = "fsl,imx6q-mipi-csi2";
 		reg = <0x021dc000 0x4000>;
 		interrupts = <0 100 0x04>, <0 101 0x04>;
-		clocks = <&clks 138>, <&clks 53>;
-		clock-names = "dphy_clk", "pixel_clk";
+		clocks = <&clks 138>, <&clks 53>, <&clks 204>;
+		clock-names = "dphy_clk", "pixel_clk", "cfg_clk";
 		status = "disabled";
 	};
 
diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index c4902e2..308a614 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -1672,13 +1672,13 @@
 				compatible = "fsl,imx6q-mipi-csi2";
 				reg = <0x021dc000 0x4000>;
 				interrupts = <0 100 0x04>, <0 101 0x04>;
-				clocks = <&clks 138>, <&clks 53>;
+				clocks = <&clks 138>, <&clks 53>, <&clks 204>;
 				/* Note: clks 138 is hsi_tx, however, the dphy_c
 				 * hsi_tx and pll_refclk use the same clk gate.
 				 * In current clk driver, open/close clk gate do
 				 * use hsi_tx for a temporary debug purpose.
 				 */
-				clock-names = "dphy_clk", "pixel_clk";
+				clock-names = "dphy_clk", "pixel_clk", "cfg_clk";
 				status = "disabled";
 			};
 
-- 
1.7.5.4

